cordformation/src/main/kotlin/net/corda/plugins/Node.kt

```diff
@@ -470,6 +491,50 @@ open class Node @Inject constructor(private val project: Project) {
         }
     }
 
+
+    private fun createSchemasCmd() = listOfNotNull(
+            Paths.get(System.getProperty("java.home"), "bin", "java").toString(),
+            "-jar",
+            "corda.jar",
+            "run-migration-scripts",
+            if (allowHibernateToManageAppSchema.get()) "--allow-hibernate-to-manage-app-schema" else null)
+
+    private fun runSchemaMigration(){
+        if (!runSchemaMigration.get()) return
+        project.logger.lifecycle("Run database schema migration scripts${if(allowHibernateToManageAppSchema.get()) " - managing CorDapp schemas with hibernate" else ""}")
+        runNodeJob(createSchemasCmd(), "node-schema-cordform.log")
+    }
+
+    val LOGS_DIR_NAME: String = "logs"
+
+    private fun runNodeJob(command: List<String>, logfileName: String) {
+        val logsDir = Files.createDirectories(nodeDir.toPath().resolve(LOGS_DIR_NAME))
+        val nodeRedirectFile = logsDir.resolve(logfileName).toFile()
+        val process = ProcessBuilder(command)
+                .directory(nodeDir)
+                .redirectErrorStream(true)
+                .redirectOutput(nodeRedirectFile)
+                .apply { environment()["CAPSULE_CACHE_DIR"] = "../.cache" }
+                .start()
+        try {
+            if (!process.waitFor(nodeJobTimeOut.get(), TimeUnit.MINUTES)) {
+                process.destroyForcibly()
+                printNodeOutputAndThrow(nodeRedirectFile)
+            }
+            if (process.exitValue() != 0) printNodeOutputAndThrow(nodeRedirectFile)
+        } catch (e: InterruptedException) {
+            // Don't leave this process dangling if the thread is interrupted.
+            process.destroyForcibly()
+            throw e
+        }
+    }
+
+    private fun printNodeOutputAndThrow(stdoutFile: File) {
```

Comment:
Perhaps also give this function a return type of `Nothing` so that Kotlin knows it will not return?